value.go

```diff
@@ -1099,3 +1106,74 @@ func (vlog *valueLog) updateDiscardStats(stats map[uint32]int64) {
 		vlog.discardStats.Update(fid, discard)
 	}
 }
+
+type vlogThreshold struct {
+	percentile     float64
+	valueThreshold int64
+	valueCh        chan ReqValSize
+	closer         *z.Closer
+	// Metrics contains a running log of statistics like amount of data stored etc.
+	vlMetrics *z.HistogramData
+}
+
+func initVlogThreshold(opt *Options) *vlogThreshold {
+	getBounds := func() []float64 {
+		mxbd := opt.maxValueThreshold
+		mnbd := float64(opt.MinValueThreshold)
+		y.AssertTruef(mxbd >= mnbd, "maximum threshold bound is less than the min threshold")
+		size := math.Min(mxbd-mnbd+1, 1024.0)
+		bdstp := (mxbd - mnbd) / size
+		bounds := make([]float64, int64(size))
+		for i := range bounds {
+			if i == 0 {
+				bounds[0] = mnbd
+				continue
+			}
+			if i == int(size-1) {
+				bounds[i] = mxbd
+				continue
+			}
+			bounds[i] = bounds[i-1] + bdstp
+		}
+		return bounds
+	}
+	return &vlogThreshold{
+		percentile:     opt.VLogPercentile,
+		valueThreshold: opt.MinValueThreshold,
+		valueCh:        make(chan ReqValSize, 1000),
+		closer:         z.NewCloser(1),
+		vlMetrics:      z.NewHistogramData(getBounds()),
+	}
+}
+
+func (v *vlogThreshold) update(sizes ReqValSize) {
+	v.valueCh <- sizes
+}
+
+func (v *vlogThreshold) close() {
+	v.closer.SignalAndWait()
+}
+
+func (v *vlogThreshold) listenForValueThresholdUpdate() {
+	defer v.closer.Done()
+	for {
+		select {
+		case <-v.closer.HasBeenClosed():
+			return
+		case val := <-v.valueCh:
+			if val == nil {
+				continue
+			}
+			for _, e := range val {
+				v.vlMetrics.Update(e)
+			}
+			// we are making it to get 99 percentile so that only values
+			// in range of 1 percentile will make it to the value log
```

Comment:
Update the comment. We are not making it 99. It depends on `opt.VlogPercentile`.

Reply:
yes thanks. It was changed recently. comment was outdated thanks :).